File: common/src/main/java/com/yekong/common/baseadapter/ItemClickListener.java
package com.yekong.common.baseadapter;

import android.view.View;

/**
 * Created by xigua on 2017/10/17.
 */

public interface ItemClickListener<M> {
    //子view点击
    void onViewClick(int id, int position, M model);
    //布局view点击
    void onRootViewClick(View view, int position, M model);
}

File: kotlin_common/src/main/java/com/yekong/kotlin_common/baseadapter/ItemClickListener.kt
package com.yekong.kotlin_common.baseadapter

import android.view.View

/**
 * Created by xigua on 2017/10/19.
 */
interface ItemClickListener <M>{
    fun onItemClick(v : View, p : Int, m : M)
    fun onViewClick(id : Int , p : Int , m : M)
}